pkg/pushsync/metrics.go

```diff
@@ -14,48 +14,138 @@ type metrics struct {
 	// to be able to return them by Metrics()
 	// using reflection
 
-	SendChunkCounter      prometheus.Counter
-	SendChunkTimer        prometheus.Counter
-	SendChunkErrorCounter prometheus.Counter
-	MarkAndSweepTimer     prometheus.Counter
+	TotalChunksToBeSentCounter prometheus.Counter
+	TotalChunksSynced          prometheus.Counter
+	TotalChunksStoredInDB       prometheus.Counter
 
-	ChunksInBatch prometheus.Gauge
+	ChunksSentCounter         prometheus.Counter
+	ChunksReceivedCounter     prometheus.Counter
+	SendChunkErrorCounter     prometheus.Counter
+	ReceivedChunkErrorCounter prometheus.Counter
+
+	ReceiptsReceivedCounter    prometheus.Counter
+	ReceiptsSentCounter        prometheus.Counter
+	SendReceiptErrorCounter    prometheus.Counter
+	ReceiveReceiptErrorCounter prometheus.Counter
+
+	ErrorSettingChunkToSynced prometheus.Counter
+	RetriesExhaustedCounter   prometheus.Counter
+	InvalidReceiptReceived    prometheus.Counter
+
+	SendChunkTimer    prometheus.Counter
+	ReceiptRTT        prometheus.Counter
```

Comment:
This should be prometheus.Histogram, I do not think what counter is useful here. Example: api package ResponseDuration metric.